### PR TITLE
ShaderGen: Toggle value of uninitialized color.

### PR DESCRIPTION
**Pull request by degasus**
SMS seems to need 0, no regressions either.

### CODE REVIEW

**Comment by Sonicadvance1** [comment]
I support this PR

File: Source/Core/VideoCommon/LightingShaderGen.h [hunk 1]
```diff
@@ -149,8 +149,8 @@ static void GenerateLightingShader(T& object, LightingUidData& uid_data, int com
 				else
 					// TODO: this isn't verified. Here we want to read the ambient from the vertex,
 					// but the vertex itself has no color. So we don't know which value to read.
-					// Returing 1.0 is the same as disabled lightning, so this could be fine
-					object.Write("lacc = int4(255, 255, 255, 255);\n");
+					// Those boxes in SMS seems to need 0.
```
Comment (JosJuice) [inline Source/Core/VideoCommon/LightingShaderGen.h]: "Those boxes in SMS" is vague. Maybe you could write something like "Boxes in Super Mario Sunshine bonus stages seem to incorrectly be visible if this isn't 0".

**Comment by delroth** [comment]
I support this PR if it gets an hwtest.

**Comment by JMC47** [comment]
Fixes a super interesting issue.  All good for me.  How do we test it on hardware?

**Comment by bb010g** [comment]
What's happening with the `inverted-depth-range` tests?

**Comment by degasus** [comment]
@bb010g I'm not sure which one is correct.

**Comment by phire** [comment]
I'm not sure we really need hardware tests for this. FifoCI results are very positive.

**Comment by roflcopter777** [comment]
Slight issue with 4.0-8388, this change completely breaks THPS3, as in causing invisible characters in-game. I reported this to the Dolphin Team but they claim to only "be my issue on my end" and yet, I tested 4.0-8385 and the issue is completely nonexistent. 

https://www.youtube.com/watch?v=r-7P9SWaWSM  This was tested with 4.0-8443, why this is happening, I don't know. I don't even think this will reach anyone or prove vital. I don't even know why I'm reporting this issue as it will only fall on deaf ears anyway. Again, because "it must be your end".

**Comment by JosJuice** [comment]
@roflcopter777 Please create an issue report at the issue tracker. Most developers don't monitor GitHub comments on closed PRs. https://bugs.dolphin-emu.org/projects/emulator

**Comment by JosJuice** [comment]
@roflcopter777 JMC made a report now, so you don't have to make a new one. You can comment on it if you want to: https://bugs.dolphin-emu.org/issues/9210

**Comment by roflcopter777** [comment]
*Sigh* Dammit, sorry -_-

**Pull request merged**